azure-redis-1shot.jsonc

```diff
@@ -0,0 +1,26 @@
+{
+    "config_path": [
+        "./mlos_bench/config",
```

Comment:
Been having to hack at these embedded paths quite a bit and there's a bunch of problems with them.

```suggestion
        "mlos_bench/config",
```

I think we should omit the leading `./` when we want the search path resolution logic to take effect and use `./` to be interpreted as either relative to the cwd where the script was executed from or relative to this file, though the latter requires additional complicated logic that I'm not sure will be great to embed in the ConfigPersistence module directly since it may need to be specialized per key.  For instance, how we refer to scripts in "environment" configs: right now they're specified as a string including "a/path/to/script plus args".  That works for shell environments, but doesn't let us identify anything about "a/path/to/script" for early validation.  Supporting alternative syntaxes like `devcontainer.json` does for array based methods like `["a/path/to/script", "plus", "args"]` could also be helpful.

Anyways, much of that discussion is not relevant here, but I figure worth bringing up for future things.


What I'd really like to do is add support for a json schemas on these files so that we can have more clearly defined semantics for expected fields in different places and easy documentation and validation, but that's also future work.

Reply:
agree. I'll remove the leading `./` now, and later we can implement all other stuff